src/components/ModalEdit/ModalEdit.tsx

```diff
@@ -32,26 +32,41 @@ const ModalEdit: FC<ModalEditProps> = ({data}: ModalEditProps) => {
     }
   };
 
+  const handleBackdrop = (event: React.FormEvent<EventTarget>): void => {
+    if (event.currentTarget === event.target) {
+      onToggleModal();
+    }
+  };
+
```

Comment:
you can use ClickAwayListener from material ui
https://material-ui.com/ru/components/click-away-listener/